src/effects/EffectChain.js

```diff
@@ -0,0 +1,74 @@
+class EffectChain {
+    constructor (audioEngine) {
+        this.audioEngine = audioEngine;
+
+        this.outputNode = this.audioEngine.audioContext.createGain();
+
+        this.lastEffect = null;
+
+        this._effects = audioEngine.effects.map(Effect => {
+            const effect = new Effect(audioEngine, this, this.lastEffect);
+            this[effect.name] = effect;
+            this.lastEffect = effect;
+            return effect;
+        });
+
+        // Walk backwards through effects connecting the last output to audio engine,
+        // then each effect's output to the input of the next effect.
+        this._effects.reduceRight((nextNode, effect) => {
+            effect.connect(nextNode);
+            return effect;
+        }, this.audioEngine);
```

Comment:
Lets make connecting to audioEngine done through a `.connect` method like Effect and SoundPlayer.

Reply:
you mean that soundbank would connect soundplayer -> effectchain -> audioEngine ?